agent.go

```diff
@@ -192,7 +192,6 @@ func (c *controller) agentSetup() error {
 	if remoteAddr != "" {
 		if err := c.agentJoin(remoteAddr); err != nil {
 			logrus.Errorf("Error in agentJoin : %v", err)
-			return nil
```

Comment:
When we return early `agentInitDone` is not closed. If it is not closed the go routing waiting on it to proceed to plumbing the ingress sandbox will not happen.